src/pages/[lang]/404.astro

```diff
@@ -0,0 +1,15 @@
+---
+import StarlightPage from '@astrojs/starlight/components/StarlightPage.astro';
+import NotFound from '~/components/NotFound.astro';
+import languages from '~/i18n/languages';
+import type { UILanguageKeys } from '~/i18n/translation-checkers';
+
+export const getStaticPaths = () =>
+	Object.keys(languages).map((lang) => ({ params: { lang: lang as UILanguageKeys } }));
+
+const { lang } = Astro.params;
```

Comment:
```suggestion
	[undefined, ...Object.keys(languages)].map((lang) => ({ params: { lang: lang as UILanguageKeys | undefined } }));

const { lang = 'en' } = Astro.params;
```
This would work if this page is renamed to `[...lang]/404.astro` and `404.astro` is removed

Reply:
Ah yeah, that’s possible, but I think sometimes it’s nicer to duplicate code a bit and keep the behaviour clear and explicit from looking at the file system.